frontend/src/pages/notebookController/StartServerModal.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import {
+  ActionList,
+  ActionListItem,
+  Button,
+  ExpandableSection,
+  List,
+  ListItem,
+  Modal,
+  ModalVariant,
+  Spinner,
+} from '@patternfly/react-core';
+import NotebookControllerContext from './NotebookControllerContext';
+
+import './NotebookController.scss';
+
+type StartServerModalProps = {
+  startShown: boolean;
+  setStartModalShown: (shown: boolean) => void;
+  onClose: () => void;
+};
+
+type SpawnStatus = {
+  status: 'success' | 'danger' | 'warning' | 'info' | 'default';
+  title: string;
+  reason: React.ReactNode;
+};
+
+const StartServerModal: React.FC<StartServerModalProps> = ({
+  startShown,
+  setStartModalShown,
+  onClose,
+}) => {
+  const { notebook, isNotebookRunning, projectName } = React.useContext(NotebookControllerContext);
+  const [spawnStatus, setSpawnStatus] = React.useState<SpawnStatus | null>(null);
+  const [isMessageExpanded, setMessageExpanded] = React.useState<boolean>(false);
+
+  // const renderMessages = (
```

Comment:
yeah, I left the dead code here for future use because we might add the logs back in the future when we have them. I can definitely remove them for now.